tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsIntegrationTest.java

```diff
@@ -119,93 +86,6 @@ public void tearDown() {
-
-        Map<String, String> oldIbpConfig = singletonMap(INTER_BROKER_PROTOCOL_VERSION_CONFIG, IBP_2_7_IV1.version());
-        Map<Integer, Map<String, String>> brokerConfigOverrides = new HashMap<>();
-        brokerConfigOverrides.put(1, oldIbpConfig);
-        brokerConfigOverrides.put(2, oldIbpConfig);
-        brokerConfigOverrides.put(3, oldIbpConfig);
-
-        cluster = new ReassignPartitionsTestCluster(Collections.emptyMap(), brokerConfigOverrides);
-        cluster.setup();
-
-        executeAndVerifyReassignment();
-    }
-
-    private void executeAndVerifyReassignment() throws ExecutionException, InterruptedException {
-        String assignment = "{\"version\":1,\"partitions\":" +
-            "[{\"topic\":\"foo\",\"partition\":0,\"replicas\":[0,1,3],\"log_dirs\":[\"any\",\"any\",\"any\"]}," +
-            "{\"topic\":\"bar\",\"partition\":0,\"replicas\":[3,2,0],\"log_dirs\":[\"any\",\"any\",\"any\"]}" +
-            "]}";
-
-        TopicPartition foo0 = new TopicPartition("foo", 0);
-        TopicPartition bar0 = new TopicPartition("bar", 0);
-
-        // Check that the assignment has not yet been started yet.
-        Map<TopicPartition, PartitionReassignmentState> initialAssignment = new HashMap<>();
-
-        initialAssignment.put(foo0, new PartitionReassignmentState(asList(0, 1, 2), asList(0, 1, 3), true));
-        initialAssignment.put(bar0, new PartitionReassignmentState(asList(3, 2, 1), asList(3, 2, 0), true));
-
-        waitForVerifyAssignment(cluster.adminClient, assignment, false,
-            new VerifyAssignmentResult(initialAssignment));
-
-        // Execute the assignment
-        runExecuteAssignment(cluster.adminClient, false, assignment, -1L, -1L);
-        assertEquals(unthrottledBrokerConfigs, describeBrokerLevelThrottles(unthrottledBrokerConfigs.keySet()));
-        Map<TopicPartition, PartitionReassignmentState> finalAssignment = new HashMap<>();
-        finalAssignment.put(foo0, new PartitionReassignmentState(asList(0, 1, 3), asList(0, 1, 3), true));
-        finalAssignment.put(bar0, new PartitionReassignmentState(asList(3, 2, 0), asList(3, 2, 0), true));
-
-        VerifyAssignmentResult verifyAssignmentResult = runVerifyAssignment(cluster.adminClient, assignment, false);
-        assertFalse(verifyAssignmentResult.movesOngoing);
-
-        // Wait for the assignment to complete
-        waitForVerifyAssignment(cluster.adminClient, assignment, false,
-            new VerifyAssignmentResult(finalAssignment));
-
-        assertEquals(unthrottledBrokerConfigs,
-            describeBrokerLevelThrottles(unthrottledBrokerConfigs.keySet()));
-
-        // Verify that partitions are removed from brokers no longer assigned
-        verifyReplicaDeleted(foo0, 2);
-        verifyReplicaDeleted(bar0, 1);
-    }
-
     @ParameterizedTest
     @ValueSource(strings = {"zk", "kraft"})
     public void testHighWaterMarkAfterPartitionReassignment(String quorum) throws Exception {
```

Comment:
It seems the check list of this case is the new leader replica has synced the HWM. Hence, we can rewrite it by following steps.

1. create a partition with 2 replicas (for example: 0,1)
2. add some records
3. create new reassignment with a new replica leader (for example: 2,0,1)
4. wait for the new leader to be on board
5. check the HWD (by `Admin#listOffsets`)